### DISCUSSION TITLE
Odd intermittent bug with icon tiling and/or text tiling instead of the base map

### DISCUSSION THREAD

**Discussion by davidemerritt** [maplibre/maplibre-gl-js#2738] [Q&A]
I'm hoping someone can help point me in the right direction on to debug a strange issue.

We have recently seen very intermittent, the odd behavior of the base map layer showing up as a set of our internal icons tiled across the map or a set of letters tiled across the map. 

In one case, it appears a base map layer has partially loaded with these icons overlayed on top of it.

<img width="1071" alt="Screenshot 2023-06-27 at 4 50 04 PM" src="https://github.com/maplibre/maplibre-gl-js/assets/3820161/dd83025a-ce17-4152-bef7-47a17450ba85">

And in the second, it appears to just be the set of english characters tiled.
<img width="1149" alt="Screenshot 2023-06-27 at 4 50 31 PM" src="https://github.com/maplibre/maplibre-gl-js/assets/3820161/4b28a7b9-4ad7-4400-9009-ab3499f6648e">

I'm not seeing any javascript errors in our exception tracking tools, but a few users have written in about this and I can't seem to figure out where to start looking. Any help in pointing me towards how to debug this would be helpful.

I am using:

maplibre-gl: 3.1.0
and 
react-map-gl: ^7.0.25

seen on web, ios & android JS implementations.

**Comment by aliflux-omo** [comment]
I saw this bug too. The characters just keep showing up on the icon if the image loading fails

**Reply by gabeschine** [reply]
In case it helps in root-causing, we've seen OPs issue with a pre-3.0.0 release as well.

**Reply by davidemerritt** [reply]
@aliflux-omo - are you seeing this with custom icons from your app, or tiled characters?

**Comment by zta6** [comment]
We’re willing to pay a contractor to fix this bug for us. If you’re interested please reach out to me at [zadam@caltopo.com](mailto:zadam@caltopo.com) and we can talk about the details!
